code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm

```diff
@@ -342,49 +351,102 @@
 	desc = "You can now transmute a tank of water and a glass shard into a bottle of eldritch water."
 	gain_text = "This is an old recipe. The Owl whispered it to me."
 	cost = 1
-	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/spell/ashen_shift)
 	required_atoms = list(/obj/structure/reagent_dispensers/watertank)
 	result_atoms = list(/obj/item/reagent_containers/glass/beaker/eldritch)
-
+	
 //	---	CURSES ---
 
-/datum/eldritch_knowledge/curse/corrosion
-	name = "Curse of Corrosion"
-	gain_text = "Cursed land, cursed man, cursed mind."
-	desc = "Curse someone for 2 minutes of vomiting and major organ damage. Using a wirecutter, a heart, and an item that the victim touched  with their bare hands."
+/datum/eldritch_knowledge/curse/alteration
+	name = "Curse Of Alteration"
+	gain_text = "Mortal bodies, prisons of flesh. Death, a release..."
+	desc = "Start an alteration ritual by transmuting a wire cutter, a hatchet and an item that the victim touched with their bare hands. Inflict a debilitating curse that will cripple your target's body for 2 minutes. Add eyes, ears, limbs or tongues to the mix to disable those organs while the curse is in effect."
 	cost = 1
-	required_atoms = list(/obj/item/wirecutters,/obj/item/organ/heart)
-	next_knowledge = list(/datum/eldritch_knowledge/mad_mask,/datum/eldritch_knowledge/spell/area_conversion)
+	required_atoms = list(/obj/item/wirecutters,/obj/item/hatchet)
 	timer = 2 MINUTES
-
-/datum/eldritch_knowledge/curse/corrosion/curse(mob/living/chosen_mob)
+	var/list/debuffs = list()
+
+/datum/eldritch_knowledge/curse/alteration/on_finished_recipe(mob/living/user, list/atoms, loc)	//the ritual completed, take the payment and apply the curse
+	//declare
+	debuffs = list()
+	var/list/extra_atoms = list()
+
+	//check variables
+	for(var/A in range(1, loc))	//this
+		var/atom/atom_in_range = A
+		if(istype(atom_in_range,/obj/item/bodypart/r_leg))
+			extra_atoms |= A
+			debuffs |= "r_leg"
+		else if(istype(atom_in_range,/obj/item/bodypart/l_leg))
+			extra_atoms |= A
+			debuffs |= "l_leg"
+		else if(istype(atom_in_range,/obj/item/bodypart/r_arm))
+			extra_atoms |= A
+			debuffs |= "r_arm"
+		else if(istype(atom_in_range,/obj/item/bodypart/l_arm))
+			extra_atoms |= A
+			debuffs |= "l_arm"
+		else if(istype(atom_in_range,/obj/item/organ/tongue))
+			extra_atoms |= A
+			debuffs |= "tongue"
+		else if(istype(atom_in_range,/obj/item/organ/eyes))
+			extra_atoms |= A
+			debuffs |= "eyes"
+		else if(istype(atom_in_range,/obj/item/organ/ears))
+			extra_atoms |= A
+			debuffs |= "ears"
+		else if(istype(atom_in_range,/obj/item/organ/liver) || istype(atom_in_range,/obj/item/organ/lungs) || istype(atom_in_range,/obj/item/organ/appendix) || istype(atom_in_range,/obj/item/organ/heart))
+			extra_atoms |= A
+			debuffs |= "organs"
```

Comment:
Use a switch statement here

Reply:
Will that check for subtypes?

Reply:
oh you're right, mb. There has to be a better way to do this though. An associated list with the organ debuff comes into mind.

Reply:
Ok. I've done it a little bit different. It checks organ/bodypart type and adapts accordingly.